checklist.js

```diff
@@ -6,21 +6,23 @@ function init(){
 	document.addEventListener("progressLoad",updateUIFromSaveData);
 	obliviondata.loadJsonData().then(()=>{
 		userdata.loadSettingsFromCookie();
-		//populate sections with json data.
-		//only display stuff that user can change.
-		const base = document.getElementById("main");
-		for(const klass of obliviondata.progressClasses){
-			const hive = obliviondata.jsondata[klass.name];
-			initMulti(hive, base,0);
-		}
-		//BAD HACK to get these specific columns to wrap
-		try{
-			document.getElementById("main_nirnroot_Outdoor_Circuit").children[0].style = "break-inside:unset";
-			document.getElementById("main_misc_Oblivion_Gates_Shut_40_Random_Gates").children[0].style = "break-inside:unset";
-			document.getElementById("main_save").children[0].style = "break-inside:unset";
-		}
-		catch{
-			debugger;
+		if(window.location.pathname.substring(window.location.pathname.lastIndexOf("/") + 1).split(".")[0] != "popout"){
```

Comment:
mmm no

Reply:
I'm not really sure how to do this differently, all I know is that the popout does work if I enclose this part in this if statement and it doesn't work at all if I don't. Is it the verbose conditional that is the problem or that the conditional interrupts the regular init()? It seems to have no negative effect on the other pages.

Reply:
I don't like changing functionality based on the window.location.pathname.
Can this go in a separate script? or common functionality extracted into another file?

Reply:
The real issue is I don't actually know enough about how this part of the code works to pull specific things out lol. Currently the tracker doesn't update unless you refresh manually anyway and it automatically mirrors the spectate/local data, so if I understood the checklist and share code better I would have a "Local" and "Shared" tracker setting as a window param for the user based on if they have their stream PC separate from their checklisting PC, and somehow the local copy would have to know when to update. The resulting code for this would likely be how you are suggesting, with a separate script that borrows relevant code from the checklist.js and share.js files.

Reply:
if user is using the webpage outside of OBS, they'll need to spectate in the OBS browser to get the data.
If the user is using the webpage in browser source to check off items, then loadProgressFromCookie() (potentially followed by recalculateProgress()) should do the trick